pgmpy/factors/discrete/DiscreteFactor.py

```diff
@@ -710,6 +710,11 @@ def copy(self):
         # because __init__ methods does that.
         return DiscreteFactor(self.scope(), self.cardinality, self.values)
 
+    def is_valid_cpd(self):
+        return np.allclose(self.to_factor().marginalize(self.scope()[:1], inplace=False).values.flatten('C'),
+                            np.ones(np.product(self.cardinality[:0:-1])),
+                            atol=0.01)
```

Comment:
I guess @abinashpanda is no longer interested in this project. Since I don't have strong idea about the tolerance value, you could either suggest a good value or modify it to whatever you see fit after this PR is merged. This shouldn't become a blocker for this PR, right?